tests/consistent.rs

```diff
@@ -0,0 +1,213 @@
+/// involves adding a custom Arbitrary instance around a couple
+/// of newtypes. That way we can respect the quickcheck size hinting
+/// and shrinking and whatnot.
+pub fn backends_are_consistent(re: &str) -> Result<u64, String> {
+    let standard_backends = vec![
+        ("bounded_backtracking_re",
+            try!(ExecBuilder::new(re)
+                    .bounded_backtracking()
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("pikevm_re",
+            try!(ExecBuilder::new(re)
+                    .nfa()
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("default_re",
+            try!(ExecBuilder::new(re)
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+    ];
+
+    let utf8bytes_backends = vec![
+        ("bounded_backtracking_utf8bytes_re",
+            try!(ExecBuilder::new(re)
+                    .bounded_backtracking()
+                    .bytes(true)
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("pikevm_utf8bytes_re",
+            try!(ExecBuilder::new(re)
+                    .nfa()
+                    .bytes(true)
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("default_utf8bytes_re",
+            try!(ExecBuilder::new(re)
+                    .bytes(true)
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+    ];
+
+    let bytes_backends = vec![
+        ("bounded_backtracking_bytes_re",
+            try!(ExecBuilder::new(re)
+                    .bounded_backtracking()
+                    .only_utf8(false)
+                    .build()
+                    .map(|exec| exec.into_byte_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("pikevm_bytes_re",
+            try!(ExecBuilder::new(re)
+                    .nfa()
+                    .only_utf8(false)
+                    .build()
+                    .map(|exec| exec.into_byte_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("default_bytes_re",
+            try!(ExecBuilder::new(re)
+                    .only_utf8(false)
+                    .build()
+                    .map(|exec| exec.into_byte_regex())
+                    .map_err(|err| format!("{}", err)))),
+    ];
+
+    Ok(try!(string_checker::check_backends(&standard_backends))
+       + try!(string_checker::check_backends(&utf8bytes_backends))
+       + try!(bytes_checker::check_backends(&bytes_backends)))
```

Comment:
Probably shouldn't be using `try!` any more. :-) We now require Rust 1.20!

Reply:
I've switched from `try!` to `?`.